lua/precognition/init.lua

```diff
@@ -417,6 +417,9 @@ end
 
 ---@param opts Precognition.PartialConfig
 function M.setup(opts)
+    if not opts then
+        opts = {}
+    end
     config = vim.tbl_deep_extend("force", default, opts or {})
```

Comment:
```suggestion
    opts = opts or {}
    config = vim.tbl_deep_extend("force", default, opts)
```
This is untested, but I think this slightly more readble and removes the duplication

Reply:
agreed :smile: